bindings/rust/standard/bench/src/harness/mod.rs

```diff
@@ -166,23 +163,30 @@ pub trait TlsConnection: Sized {
 
     fn handshake_completed(&self) -> bool;
 
+    /// Send application data to ConnectedBuffer
+    fn send(&mut self, data: &[u8]) -> Result<(), Box<dyn Error>>;
+
+    /// Read application data from ConnectedBuffer
+    fn recv(&mut self, data: &mut [u8]) -> Result<(), Box<dyn Error>>;
+
+    /// shutdown send
+    fn send_shutdown(&mut self);
+    fn shutdown_completed(&mut self) -> bool;
+}
+
+pub trait TlsMetrics: Sized {
```

Comment:
Looking at the current methods here, is "metrics" the right name? Maybe "Info" or "Summary" or "Results".